tunnel/utils/ipcalc.go

```diff
@@ -17,19 +17,29 @@
 package utils
 
 import (
+	"bytes"
 	"github.com/michaelquigley/pfxlog"
 	"github.com/pkg/errors"
 	"net"
 )
 
+func IsLocallyAssigned(addr, lower, upper net.IP) bool {
+	return bytes.Compare(addr.To16(), lower.To16()) >= 0 && bytes.Compare(addr.To16(), upper.To16()) <= 0
+}
+
 // return the next available IP address in the range of provided IPs
 func NextIP(lower, upper net.IP) (net.IP, error) {
 	usedAddrs, err := AllInterfaceAddrs()
 	if err != nil {
 		return nil, err
 	}
 
```

Comment:
We were always returning `lower` from this call, since lower is just a byte array. Additionally we were mutating lower as we were operating. This meant that each new service would end up change the value for all other intercept addresses, which would cause cleanup to fail.